hudi-common/src/main/java/org/apache/hudi/common/model/HoodieFileGroup.java

```diff
@@ -62,9 +60,7 @@
   public HoodieFileGroup(HoodieFileGroup fileGroup) {
     this.timeline = fileGroup.timeline;
     this.fileGroupId = fileGroup.fileGroupId;
-    this.fileSlices = new TreeMap<>(fileGroup.fileSlices.entrySet().stream()
-        .map(e -> Pair.of(e.getKey(), new FileSlice(e.getValue())))
-        .collect(Collectors.toMap(Pair::getKey, Pair::getValue)));
+    this.fileSlices = new TreeMap<>(fileGroup.fileSlices);
```

Comment:
would you please write an unit test to cover the change?

Reply:
done